text/0000-Component-Manager-Bounds.md

```diff
@@ -0,0 +1,177 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Component Manager Bounds
+
+## Summary
+
+This is a follow up custom components RFC that outlines how we expose a component's DOM structure and associated semantics.
+
+## Motivation
+Historically, classic Ember components have had a built in escape hatch for doing DOM manipulation via `didInsertElement`. This allowed for components to encapsulate some 3rd party JavaScript library or do some fine grain DOM manipulation. We feel that is ability to drop down to a lower level abstraction makes components powerful.
+
+Today, the custom components are not given direct access to the DOM structure; internally known as `Bounds`. This limitation means that custom components cannot easily encapsulate functionality in the same way that classic Ember components can. Expanding the capabilities of custom components to expose the `Bounds` is in order to increase the usefulness of the abstraction.
+
+## Detailed design
+
+In [RFC#2013](https://github.com/emberjs/rfcs/blob/master/text/0213-custom-components.md#detailed-design) we outlined the component lifecycle and how a custom component is associated with a manager. This design builds off off of that existing work and concepts.
+
+### `elementHook` capability
+
+The primary way of introducing new functionality to the component manager is through capabilities. To allow custom component managers to announce that they support element related hooks, we will introduce an `elementHook` flag to the capabilities. In doing so, bounds will be dispatched to a `didRenderLayout` method on the manager and `willDestroyLayout` will be called during destruction.
+
+```js
+import { capabilities } from '@ember/component';
+import EmberObject from '@ember/object';
+
+export default EmberObject.extend({
+  capabilities: capabilities('3.5', {
+    elementHook: true
+  }),
+
+  // ...
+});
+```
+
+### `didRenderLayout` hook
+Once the rendering engine has constructed the bounds for the component the `didRenderLayout` hook will be called with the `component` and a `Bounds` instance.
```

Comment:
This phrasing is slightly off, the hook does not receive "the component" it receives the opaque result of calling `createComponent` (aka the "state bucket"). Different component managers _may_ return the instance itself but it may also return some other POJO/state bucket.